common/lib/client/auth.js

```diff
@@ -534,6 +519,144 @@ var Auth = (function() {
 		}
 	};
 
+	Auth.prototype._saveBasicOptions = function(defaultTokenParams, authOptions) {
+		this.method = 'basic';
+		this.key = authOptions.key;
+		this.basicKey = toBase64(authOptions.key);
+		this.tokenParams = defaultTokenParams || {};
+		this.authOptions = authOptions || {};
+		delete this.authOptions.force;
+		if('clientId' in authOptions) {
+			this._userSetClientId(authOptions.clientId);
+		}
+	}
+
+	Auth.prototype._saveTokenOptions = function(tokenParams, authOptions) {
+		this.method = 'token';
+
+		this.tokenParams = tokenParams || this.tokenParams || {};
+
+		/* If an authOptions object is passed in that contains new auth info (ie
+		* isn't just {force: true} or something), it becomes the new default, with
+		* the exception of the force attribute (RSA10g), which is set anew on each
+		* call to authorise (defaulting to false) */
+		this.force = false;
+		if(authOptions) {
+			this.force = authOptions.force;
+
+			if(containsAuthInfo(authOptions)) {
+				this.authOptions = authOptions || {};
+				delete this.authOptions.force;
+			} else if('queryTime' in authOptions) {
+				/* queryTime isn't an authInfo so can't replace the current
+				* authOptions, but per RSA10g, should still be stored */
+				this.authOptions.queryTime = authOptions.queryTime;
+			}
+
+			if(this.force) {
+				/* get rid of current token even if still valid */
+				this.tokenDetails = null;
+			}
+
+			if(authOptions.token) {
+				/* options.token may contain a token string or, for convenience, a TokenDetails */
+				authOptions.tokenDetails = (typeof(authOptions.token) === 'string') ? {token: authOptions.token} : authOptions.token;
+			}
+			if(authOptions.tokenDetails) {
+				this.tokenDetails = authOptions.tokenDetails;
+			}
+
+			if('clientId' in authOptions) {
+				this._userSetClientId(authOptions.clientId);
+			}
+		}
+	};
+
+	Auth.prototype._logTokenAuthMethod = function() {
```

Comment:
This function does not need to be exposed as part of the prototype and instead could be called with `authOptions` as an argument.  The less exposed functions on the object the better IMHO